doc/source/whatsnew/v1.4.0.rst

```diff
@@ -518,6 +518,7 @@ Performance improvements
 - Performance improvement in :meth:`Series.to_frame` (:issue:`43558`)
 - Performance improvement in :meth:`Series.mad` (:issue:`43010`)
 - Performance improvement in :func:`merge` (:issue:`43332`)
+- Performance improvement in :func:`read_csv` when ``index_col`` was set with a numeric column (:issue:`44158`)
```

Comment:
i think this is a regression compared to 1.3.x. would be ok backporting.

Reply:
This is equally slow on 1.2.5 and 1.3.4, so don't think we should backport

Reply:
oh ok